### PR TITLE
Clear SelectBox filter after the widget focusout event (T838753)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: js/ui/select_box.js [hunk 1]
```diff
@@ -590,11 +590,19 @@ var SelectBox = DropDownList.inherit({
...
+            if(this._isEditable() && !this._isOverlayNestedTarget(e.relatedTarget)) {
+                this._searchCanceled();
+            }
         }
 
         this.callBase(e);
     },
 
+    _isOverlayNestedTarget: function(target) {
+        return !!(this.content() && $(this.content()).parent().find(target).length);
```
Comment (San4es) [inline js/ui/select_box.js]:
```
$(target).closest(`.${SELECTBOX_POPUP_WRAPPER_CLASS}`).length
```
?

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 2]
```diff
@@ -2646,6 +2646,24 @@ QUnit.module('search', moduleSetup, () => {
...
+            searchEnabled: true
+        });
+
+        const instance = $selectBox.dxSelectBox('instance');
+        const $input = $selectBox.find(toSelector(TEXTEDITOR_INPUT_CLASS));
+
+        keyboardMock($input).type('1');
+        $input.trigger('focusout');
+
+        assert.equal($(instance.content()).find('.dx-list-item').length, 3, 'filter was cleared');
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]: You can use `LIST_ITEM_CLASS` here.

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 3]
```diff
@@ -4074,7 +4095,6 @@ QUnit.module('applyValueMode = \'useButtons\'', {
             .type('c');
 
         $('.dx-list-select-all-checkbox').trigger('dxclick');
-        $($input).trigger('focusout'); // Emulating the real behavior
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js:4077]: Why are we going to remove this?

**Review by San4es — APPROVED** [review]
LGTM

**Review by ksercs — APPROVED** [review]

**Pull request merged**